### PR TITLE
Add a file for documenting fullerite development

### PR DESCRIPTION
**Pull request by gnufied**

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+## Running Fullerite locally
+
+Running `make` command above should result in fullerite binary in `bin` directory.
+You can run fullerite using `bin/fullerite` command.
+
+However - typically fullerite also requires some configuration before it can
+run properly. You can look into example configuration file in `examples/config/fullerite.conf.example`
+and adjust it according to your needs.
+
+A more shorter configuration is:
```
Comment (tiras-j) [inline DEVELOPMENT.md]: Perhaps: "A simple configuration is"?

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    "path": "/metrics"
+  },
+  "collectors": ["Diamond"],
+  "handlers": {
+    "Log": {}
+  }
+}
+```
+
+The most important bits here is `collectors` and `diamondCollectors` keys. We will talk about them
```
Comment (tiras-j) [inline DEVELOPMENT.md]: "bits here is" -> "bits here are"

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+later in this document.
+
+Once you have created the configuration file - lets save it in root of fullerite directory with the name `fullerite.conf`.
```
Comment (tiras-j) [inline DEVELOPMENT.md]: "lets" -> "let's"

File: DEVELOPMENT.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+~> ./bin/fullerite -c fullerite.conf
+```
+
+
+## Adding a collector
+
+A collector in fullerite can be written using `go` programming language
+or can be written in `Python`.
+
+### Writing a collector in Go
```
Comment (tiras-j) [inline DEVELOPMENT.md]: Add specific details on the init factory function registration for new collectors?

File: DEVELOPMENT.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+You also need to create a configuration file for particular collector even if it is just `{}`. Assuming
+you are following along with example configuration above - create a file called `NewCollector.conf`
+in `examples/config/` directory.
+
+And now you can go ahead and run fullerite as documented above.
+
+### Adding a collector in Python
+
+If you are already familiar with Python - it is very easy to write a collector in Python.
+For this example - lets try to run CPU collector located at `diamond/collectors/cpu/cpu.py`.
```
Comment (tiras-j) [inline DEVELOPMENT.md]: "lets" -> "let's"

File: DEVELOPMENT.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+First you need to modify `fullerite.conf` file created above and make sure that this collector
+is enabled:
+
+```
+"diamondCollectors": [ "CPUCollector" ]
+```
+
+You also need to define a configuration file for this collector even if its contents are just `{}`.
+If you are running from Fullerite clone - `examples/config` director should already contain `CPUCollector.conf`
```
Comment (tiras-j) [inline DEVELOPMENT.md]: directory?

**Comment by tiras-j** [comment]
Really nice, succinct but gets all the important points across. Nicely done.

File: DEVELOPMENT.md (hunk 4 above)
Comment (gnufied) [inline DEVELOPMENT.md]: I didn't wanted to go too much in detail about - writing collector code itself. I suppose, people can learn that from existing code.

Running and testing collectors however have been somewhat of a pain point, so I am trying to address them here.

Comment (tiras-j) [inline DEVELOPMENT.md]: Yea good point, I just worry that the pre-registration of the factory functions isn't very straightforward. But someone who is taking the time to write a new collector should likely be capable of reading the current code lol.

**Comment by dichiarafrancesco** [comment]
I think it contains the main points. +1

**Pull request merged**